spec/services/place_hold_results/builder_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
```

Comment:
Should we add a spec to test that the loop is being executed too? Like you did for #147?

Reply:
if you mean the records in use workaround, I am thinking that we test it with symphony_client_spec, and results just makes calls to symphony_client methods so kinda leaning towards to no need to that here again but would like to hear your reasoning, I might be missing sth.

Reply:
So the workaround in #147 relies on `SymphonyClient::authenticated_request` to delay if `records_in_use?`. This one is different. It is checking for a malformed response from the WS. Then it will delay here, in this class (not in SymphonyClient::autheticated_request`. We are not testing a scenario where this results builder encounters malformed response to looking up a hold. 

Strange to me that CC is saying that the line inside the `while` loops is covered, so maybe I'm mis-reading.

Reply:
ahhhh ok yeah, the spec for that is still at holds_controller_spec https://github.com/psu-libraries/myaccount/blob/151-refactor-results/spec/controllers/holds_controller_spec.rb#L367, I will try to move it over

Reply:
Ah that's why some confusion. Hope it's an easy port over. Thanks!

Reply:
yes sorry, I apparently forgot to update these, working on it now :)

Reply:
No worries!!

Reply:
ok, pushed a commit for this change, hope it looks right